Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide/Internal.hs

```diff
@@ -97,7 +98,7 @@ cancelRideImpl rideId bookingCReason = do
     transpConf <- QTC.findByMerchantId merchant.id >>= fromMaybeM (TransporterConfigNotFound merchant.id.getId)
     let searchRepeatLimit = transpConf.searchRepeatLimit
     now <- getCurrentTime
-    farePolicy <- QFP.findByMerchantIdAndVariant searchReq.providerId searchTry.vehicleVariant >>= fromMaybeM NoFarePolicy
+    farePolicy <- getFarePolicyForVariant searchReq.providerId (LatLong searchReq.fromLocation.lat searchReq.fromLocation.lon) (LatLong searchReq.toLocation.lat searchReq.toLocation.lon) searchTry.vehicleVariant
```

Comment:
this can be a point query directly to farePolicy table after adding farePolicyId to searchTry.

Reply:
Can I do this as Next PR ?

Reply:
Done Resolved.